src/main/java/mate/academy/spring/service/impl/RoleServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package mate.academy.spring.service.impl;
+
+import java.util.NoSuchElementException;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.model.Role;
+import mate.academy.spring.service.RoleService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RoleServiceImpl implements RoleService {
+    private final RoleDao roleDao;
+
+    public RoleServiceImpl(RoleDao roleDao) {
+        this.roleDao = roleDao;
+    }
+
+    @Override
+    public Role add(Role role) {
+        return roleDao.add(role);
+    }
+
+    @Override
+    public Role getByName(Role.RoleName roleName) {

```

Comment:
Sorry? probably that was not clear enough from previous review, the service method should have String as parameter. We should work with `Role.RoleName` just on dao layer according the task description and common mistakes.